src/test/java/pl/karolpat/service/ParkingMeterServiceTest.java

```diff
@@ -0,0 +1,194 @@
+package pl.karolpat.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.when;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import pl.karolpat.entity.ParkingMeter;
+import pl.karolpat.entity.User;
+import pl.karolpat.repository.ParkingMeterRepo;
+import pl.karolpat.serviceImpl.ParkingMeterServiceImpl;
+
+@RunWith(SpringRunner.class)
+public class ParkingMeterServiceTest {
+
+	private ParkingMeterRepo parkingMeterRepo;
+	private ParkingMeterService parkingMeterService;
+	ParkingMeter parkingMeter1;
+	ParkingMeter parkingMeter2;
+	User user1;
+	String HOURS;
+	String PLN_CURRENCY;
+	Timestamp start;
+
+	@Before
+	public void setUp() throws Exception {
+		parkingMeterRepo = Mockito.mock(ParkingMeterRepo.class);
+		parkingMeterService = new ParkingMeterServiceImpl(parkingMeterRepo);
+		parkingMeter1 = new ParkingMeter();
+		parkingMeter2 = new ParkingMeter();
+		user1 = new User();
+
+		Calendar calendar = Calendar.getInstance();    //This block is to create time that will
+		calendar.setTime(new Date());					//be constant for tests to eliminate
+		calendar.add(Calendar.HOUR, -5);				//failures caused by difference in current time.
+		start = new Timestamp(calendar.getTimeInMillis());
+
+	}
+
+	@Test
+	public void testSave() {
+		// given
+		when(parkingMeterRepo.save(parkingMeter1)).thenReturn(parkingMeter1);
+		// when
+		ParkingMeter result = parkingMeterService.save(parkingMeter1);
+		// then
+		assertThat(result).isInstanceOf(ParkingMeter.class);
+		assertEquals(parkingMeter1, result);
+	}
+
+	@Test
+	public void testFindUserParkingMeter() {
+
+		user1.setId(1l);
+		parkingMeter1.setStart(Timestamp.valueOf("2017-1-1 12:55:00"));
+		parkingMeter2.setStart(Timestamp.valueOf("2017-1-1 12:55:30"));
+		parkingMeter1.setUser(user1);
+		parkingMeter2.setUser(user1);
+
+		when(parkingMeterRepo.getFirstByUserIdOrderByStart(1l)).thenReturn(parkingMeter2);
+
+		ParkingMeter result = parkingMeterService.findUserParkingMeter(1l);
+
+		assertThat(result).isInstanceOf(ParkingMeter.class);
```

Comment:
result jest ParkingMeter - po co sprawdzać jeszcze raz?

Reply:
No tak, w tym przypadku rzeczywiście jest to zbędne.